src/main/java/org/mutabilitydetector/checkers/MutableTypeToFieldChecker.java

```diff
@@ -95,8 +96,12 @@ public void visitFormalTypeParameter(String name) {
     @Override
     public FieldVisitor visitField(int access, String name, String desc, String signature, Object value) {
         if (signature == null) { return null ; }
+        if(signature.contains(this.ownerClass())) {
+            this.setIsClassSelfReferenced(true);
```

Comment:
- No need for `this.` here.
- Without further knowing the code: are you sure the `.contains` check is robust for any kind of combination and class and method signature? (Taking into account pathological cases such as a two FQCNs where one is a suffix of the other.)

Reply:
Removed the "this" reference, but just for curiosity is there any impact of using it or just for the aesthetics ?
Nice 2nd point ! I tested the for inner class references for the field as well,  the contains check did not affect the behavior of the code. I mean the code gives correct results. so it is safe i believe.

Reply:
Indeed, I suggested dropping `this.` only for consistency with the surrounding code. :)